### PR TITLE
[Indexing] Fix #43 & Copy DataFrame for view

### LINKED ISSUE #43: TypeError: '>' not supported between instances of 'list' and 'int' in iloc
### System information
- **OS Platform and Distribution**: "Ubuntu 18.04 LTS"
- **Modin installed from (source or binary)**: binary
- **Modin version**: 0.1.0
- **Python version**: 3.6.5
- **Exact command to reproduce**: df.iloc[0:5,:]

Calling: **df.iloc[0:5,:]** results in the following error:

Traceback (most recent call last):
  File "bakeoffModin.py", line 15, in <module>
    print( f'Sample:\n {df.iloc[0:5,:]} ')
  File "/usr/local/lib/python3.6/dist-packages/modin/pandas/dataframe.py", line 230, in __str__
    return repr(self)
  File "/usr/local/lib/python3.6/dist-packages/modin/pandas/dataframe.py", line 325, in __repr__
    return repr(self._repr_helper_())
  File "/usr/local/lib/python3.6/dist-packages/modin/pandas/dataframe.py", line 235, in _repr_helper_
    return to_pandas(self)
  File "/usr/local/lib/python3.6/dist-packages/modin/pandas/utils.py", line 227, in to_pandas
    pandas_df = pandas.concat(ray.get(df._row_partitions), copy=False)
  File "/usr/local/lib/python3.6/dist-packages/modin/pandas/dataframe.py", line 198, in _get_row_partitions
    empty_rows_mask = self._row_metadata._lengths > 0
TypeError: '>' not supported between instances of 'list' and 'int'

### PR DESCRIPTION
**Pull request by simon-mo**
## What do these changes do?
Replace PR #45. This PR make less breaking changes. 

## Related issue number
#43

- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by simon-mo** [comment]
`df.iloc[0:5, :]` is going to be "Dataframe View". And the repr for this is currently broken. It will take some time to fix it.

File: modin/pandas/utils.py [hunk 1]
```diff
@@ -263,6 +263,85 @@ def to_pandas(df):
...
+            oid = extract_block.remote(
+                block_partition_2d_oid_arr,
+                row_lookup,
+                col_lookup,
+                col_name_suffix='_old')
+            oid_lst.append(oid)
+    return np.array(oid_lst).reshape(len(row_grouped), len(col_grouped))
+
+
+# Indexing
```
Comment (devin-petersohn) [inline modin/pandas/utils.py]: What are these used for? We haven't used this convention elsewhere, do you think we should?

Comment (simon-mo) [reply modin/pandas/utils.py]: We are going to refractor utils into a package, and these functions will be in separate files named under `indexing`, and this comment block will be part of the file level description.

File: modin/pandas/utils.py [hunk 2]
```diff
@@ -263,6 +263,85 @@ def to_pandas(df):
...
+        {
+            'partition': partitions,
+            'index_within_partition': idx_in_part
+        },
+        index=old_coord_df.index)
+
+    return final_df
+
+
+def _generate_blocks(old_row, new_row, old_col, new_col,
```
[the remark this answers is not in the document]
Comment (simon-mo) [reply modin/pandas/utils.py]: added

**Comment by simon-mo** [comment]
@devin-petersohn done!

File: modin/pandas/utils.py [hunk 3]
```diff
@@ -433,11 +512,11 @@ def _deploy_func(func, dataframe, *args):
 def extractor(df_chunk, row_loc, col_loc):
     """Retrieve an item from remote block
     """
-    # We currently have to do the writable flag trick because a pandas bug
+    # We currently have to do this hack because a pandas bug
     # https://github.com/pandas-dev/pandas/issues/17192
     try:
-        row_loc.flags.writeable = True
-        col_loc.flags.writeable = True
+        row_loc = row_loc.copy()
```
[the remark this answers is not in the document]
Comment (devin-petersohn) [reply modin/pandas/utils.py]: Sorry, what I meant was to change this back to `row_loc.flags.writeable = True` and keep the extra except.

File: modin/pandas/utils.py (hunk 1 above)
Comment (devin-petersohn) [reply modin/pandas/utils.py]: Sounds good. I like the idea of using these and have adopted the general concept in the rewrite.

File: modin/pandas/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (devin-petersohn) [reply modin/pandas/utils.py]: Great thanks!

**Pull request merged**